NullReferenceException in NavigationService

**Android API:** Android 8.1
**Mapbox Navigation SDK version:** Navigation Library 0.13.0

### Steps to trigger behavior

At this stage, unsure what triggered it, but I've seen it more than once. It might be if startNavigation() is never called (in which case the service will never set the mapboxNavigation object)

### Expected behavior

### Actual behavior

```
05-04 11:12:56.372 2401-2401/com.bynd.looper E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.bynd.looper, PID: 2401
    java.lang.RuntimeException: Unable to stop service com.mapbox.services.android.navigation.v5.navigation.NavigationService@bf2966d: java.lang.NullPointerException: Attempt to invoke virtual method 'com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationOptions com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation.options()' on a null object reference
        at android.app.ActivityThread.handleStopService(ActivityThread.java:3735)
        at android.app.ActivityThread.-wrap30(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1741)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6682)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationOptions com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation.options()' on a null object reference
        at com.mapbox.services.android.navigation.v5.navigation.NavigationService.onDestroy(NavigationService.java:85)
        at android.app.ActivityThread.handleStopService(ActivityThread.java:3718)
        at android.app.ActivityThread.-wrap30(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1741) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6682) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410) 
```

--- a/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java
+++ b/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java
@@ -82,7 +82,7 @@ public class NavigationService extends Service implements LocationEngineListener
 
   @Override
   public void onDestroy() {
-    if (mapboxNavigation.options().enableNotification()) {
+    if (isValidOptions() && mapboxNavigation.options().enableNotification()) {
       stopForeground(true);
     }
     endNavigation();
@@ -110,7 +110,7 @@ public class NavigationService extends Service implements LocationEngineListener
    */
   @Override
   public void onNewRouteProgress(Location location, RouteProgress routeProgress) {
-    if (mapboxNavigation.options().enableNotification()) {
+    if (isValidOptions() && mapboxNavigation.options().enableNotification()) {
       navigationNotification.updateNotification(routeProgress);
     }
     mapboxNavigation.getEventDispatcher().onProgressChange(location, routeProgress);
@@ -273,13 +273,12 @@ public class NavigationService extends Service implements LocationEngineListener
     startForeground(notificationId, notification);
   }
 
-  /**
-   * Runs several checks on the actual rawLocation object itself in order to ensure that we are
-   * performing navigation progress on a accurate/valid rawLocation update.
-   */
-  @SuppressWarnings("MissingPermission")
   private boolean isValidLocationUpdate(Location location) {
     return location != null && locationValidator.isValidUpdate(location);
+  }
+
+  private boolean isValidOptions() {
+    return mapboxNavigation != null && mapboxNavigation.options() != null;
   }
 
   /**

